src/main/java/net/sacredlabyrinth/phaed/dynmap/simpleclans/Preferences.java

```diff
@@ -0,0 +1,95 @@
+package net.sacredlabyrinth.phaed.dynmap.simpleclans;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import net.sacredlabyrinth.phaed.simpleclans.Clan;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+
+public class Preferences {
+    private static final File prefFile = new File(DynmapSimpleClans.getInstance().getDataFolder(), "preferences.json");
+    private static JsonArray preferences;
+    private final Clan clan;
+
+    public Preferences(@NotNull Clan clan) {
+        this.clan = clan;
+        if (preferences == null) {
+            loadPreferences();
+        }
+    }
+
+    public static void loadPreferences() {
```

Comment:
Does it need to be public?
Is it called in the main thread?

Reply:
https://github.com/Tomut0/Dynmap-SimpleClans/blob/09691a60e1bf8b2a16d098a562b84193dcef0a62/src/main/java/net/sacredlabyrinth/phaed/dynmap/simpleclans/DynmapSimpleClans.java#L88